code/game/objects/items/weapons/grenades/chem_grenade.dm

```diff
@@ -349,8 +349,11 @@
 					for(var/obj/item/reagent_containers/glass/G in beakers)
 						S.reagents.trans_to(G, S.reagents.total_volume)
 				else
-					S.forceMove(get_turf(src))
 					no_splash = TRUE
+					if(beakers.len)
+						for(var/obj/item/slime_extract/O in beakers)
+							O.forceMove(get_turf(src))
+							beakers = list()
```

Comment:
why are you setting the list inside of the loop
also why isn't the one at line 352 good enough - is the trouble that slime cores without uses are getting skipped? Is the trouble that the cores aren't being removed from the `beakers` list?